### DISCUSSION TITLE
Time to Live TTL can be used to update the new visibility timeout value feature ?

### DISCUSSION THREAD

**Discussion by padhumailin** [Azure/azure-cosmos-dotnet-v3#2984] [Q&A]
Hi

while update nuget package from Microsoft.Azure.DocumentDB.Core to Microsoft.Azure.Cosmos. 
We have existing method in queueclient to update the visibility time duration, in case of failed queue process
`Task<Response<UpdateReceipt>> UpdateMessageAsync(string messageId, string popReceipt, string messageText = null, TimeSpan visibilityTimeout = default, CancellationToken cancellationToken = default);`

to migrate this method, i have seen TTL (time to live) in Microsoft.Azure.Cosmos. 
does these methods meant for same purpose?

https://docs.microsoft.com/en-us/azure/cosmos-db/sql/how-to-time-to-live?tabs=dotnetv3%2Cjavav3#reset-time-to-live

`// This examples leverages the Sales Order class above.
// Read a document, update its TTL, save it.
ItemResponse<SalesOrder> itemResponse = await client.GetContainer("database", "container").ReadItemAsync<SalesOrder>("SO05", new PartitionKey("CO18009186470"));

itemResponse.Resource.ttl = 60 * 30 * 30; // update time to live
await client.GetContainer("database", "container").ReplaceItemAsync(itemResponse.Resource, "SO05");`

can i use ReplaceItemAsync method in place of UpdateMessageAsync?

**Comment by kirankumarkolli** [comment]
Microsoft.Azure.DocumentDb.Core also doesn't have UpdateDocument, it only has ReplaceDocument API's. 
If your earlier implementation is using ReplaceDocument then ReplaceItem API's works. 

Please note that Replace will fail if the Item doesn't exist already.
